_posts/2024-04-15-heapdump-oome-native.adoc

```diff
@@ -0,0 +1,258 @@
+---
+layout: post
+title: 'Obtaining heap dump on OutOfMemoryError with Quarkus native'
+date: 2024-04-15
+tags: native heapdump oome epsilongc jol
+synopsis: 'Generating and inspecting heap dumps in Quarkus native when `OutOfMemoryError` errors occur'
+author: galderz
+---
+
+Starting with GraalVM for JDK 21,
+native executables can run with the `-XX:+HeapDumpOnOutOfMemoryError` option to generate a heap dump when a `java.lang.OutOfMemoryError` is thrown.
+In this blog post we will explore how to use the flag, we will inspect what a native image heap dump looks like and how it compares with one produced by HotSpot.
```

Comment:
>we will inspect what a native image heap dump looks like

Maybe there's a chance reader's  get confused by the wording here and think dumping the  Image Heap and "java heap" are different things (If they are aware that the image heap exists).

Maybe it would be good to specify that we're dumping both the read-only Image Heap as well as the native image's "Java heap". Whereas in Java mode we're just dumping the Java heap.

Reply:
On the flip side, it could also be an opportunity to briefly inform reader's who aren't aware that there is a special segmentation to the heap in native mode.

Reply:
Both points sound good. I have not explored the differences in detail between the read-only and the read/write parts, but if you have feel free to propose a paragraph and add it as suggestion and we can just add it.

Reply:
ok I've added a suggestion below!